cirq/ops/moment.py

```diff
@@ -116,6 +116,32 @@ def with_operation(self, operation: 'cirq.Operation') -> 'cirq.Moment':
 
         return m
 
+    def with_operations(self, *contents: 'cirq.OP_TREE') -> 'cirq.Moment':
```

Comment:
This is equivalent to what the `__add__` operation does.

It's probably reasonable to have an autocomplete-discoverable name for it. But mention the `__add__` approach as an alternative, and delegate it instead of duplicating code.

Reply:
Done. I kept the code in `with_operations` and made `__add__` delegate to this since it doesn't re-index the existing operations and so will be more efficient in the case of adding a small additional number of operations into a large moment. Also kept the special behavior of `__add__` of delegating to `Circuit.__radd__` when adding to a circuit, which we don't want for `with_operations`.